src/providers/ConnectorState/ConnectorLogsSlice.ts

```diff
@@ -2,31 +2,55 @@ import { createSlice, PayloadAction } from '@reduxjs/toolkit'
 
 import type { RootState } from '/store'
 
-interface ConnectorLogsState {
-  logs: string[]
+export interface LogObj {
+  msg: string
+  timestamp: string
+  level: string
+  slug: string
+}
+
+type LogDict = Record<string, LogObj[] | undefined>
+
+export interface ConnectorLogsState {
+  logs: LogDict
+}
+
+export interface removeLogInfo {
+  slug: string
+  number: number
 }
 
 const initialState: ConnectorLogsState = {
-  logs: []
+  logs: {}
 }
 
 export const connectorLogsSlice = createSlice({
   name: 'connectorLogs',
   initialState,
   reducers: {
-    addLog: (state, action: PayloadAction<string>) => {
-      state.logs = [...state.logs, action.payload]
+    addLog: (state, action: PayloadAction<LogObj>) => {
+      if (state.logs[action.payload.slug] === undefined) {
+        state.logs[action.payload.slug] = []
+      }
+      state.logs[action.payload.slug]?.push(action.payload)
     },
-    spliceLogs: (state, action: PayloadAction<number>) => {
-      state.logs = state.logs.slice(action.payload)
+    removeLogs: (state, action: PayloadAction<removeLogInfo>) => {
+      const result = state.logs[action.payload.slug]?.slice(
+        action.payload.number
+      )
+      if (result?.length) {
+        state.logs[action.payload.slug] = result
+      } else {
+        delete state.logs[action.payload.slug]
```

Comment:
delete is extremely slow in javascript. Setting undefined is way better. 

You can find a lot of readings about that on the internet, here is one of the first link I found https://webkit.org/blog/10298/inline-caching-delete/